File: DataModule/src/com/mobnetic/coinguardian/util/MarketsConfigUtils.java
package com.mobnetic.coinguardian.util;

import java.util.ArrayList;

import com.mobnetic.coinguardian.config.MarketsConfig;
import com.mobnetic.coinguardian.model.Market;
import com.mobnetic.coinguardian.model.market.Unknown;

public class MarketsConfigUtils {

	private final static Market UNKNOWN = new Unknown();
	
	public static Market getMarketById(int id) {
		synchronized (MarketsConfig.MARKETS) {
			if(id>=0 && id<MarketsConfig.MARKETS.size()) {
				return new ArrayList<Market>(MarketsConfig.MARKETS.values()).get(id);
			}
		}
		return UNKNOWN;
	}
	
	public static Market getMarketByKey(String key) {
		synchronized (MarketsConfig.MARKETS) {
			if(MarketsConfig.MARKETS.containsKey(key))
				return MarketsConfig.MARKETS.get(key);
		}
		return UNKNOWN;
	}
	
	public static int getMarketIdByKey(String key) {
		int i=0;
		for(Market market : MarketsConfig.MARKETS.values()){
			if(market.key.equals(key))
				return i;
			++i;
		}
		
		return 0;
	}
}

File: dataModule/src/main/java/com/aneonex/bitcoinchecker/datamodule/util/MarketsConfigUtils.kt
package com.aneonex.bitcoinchecker.datamodule.util

import com.aneonex.bitcoinchecker.datamodule.config.MarketsConfig
import com.aneonex.bitcoinchecker.datamodule.model.Market
import com.aneonex.bitcoinchecker.datamodule.model.market.UnknownMarket

object MarketsConfigUtils {
    private val UNKNOWN_MARKET: Market = UnknownMarket()

    @Suppress("unused")
    val defaultMarket: Market = MarketsConfig.MARKETS.values.first()

    fun getMarketByKey(key: String?): Market {
        synchronized(MarketsConfig.MARKETS) {
            return MarketsConfig.MARKETS.getOrDefault(key, null) ?: UNKNOWN_MARKET
        }
    }
/*
    fun getMarketIdByKey(key: String): Int {
        for ((i, market) in MarketsConfig.MARKETS.values.withIndex()) {
            if (market.key == key) return i
        }
        return 0
    }
 */
}